apps/api/src/app/logo/logo.controller.ts

```diff
@@ -24,19 +24,26 @@ export class LogoController {
   // TODO: create an interface for the body type like "@Body() validate: ValidationPayload"
   @Post(':id/validate')
   @UseGuards(JwtAuthGuard)
-  async validateGuess(@Param('id') id: string, @Body() validate: { guess: string }, @Req() request: Request) {
+  async validateGuess(
+    @Param('id') id: string,
+    @Body() validate: { guess: string },
+    @Req() request: Request): Promise<LogoVerifyResponse> {
     const guess = validate.guess;
     const logo = await this.logoService.findOne(id);
-    const status = logo.name === guess;
+    const logoObject = logo.toJSON() as Logo;
+    const status = logoObject.name === guess;
     const user = request['user'];
     const state = await this.userStateService.findByUser(user.id);
     if (status) {
-      const isValidated = state.logos.indexOf(logo.id) !== -1;
+      const isValidated = state.logos.indexOf(logoObject.id) !== -1;
       if (!isValidated) {
         await this.userStateService.insertLogo(user.id, logo);
       }
     }
-    return { status };
+    return {
+      status,
+      realImageUrl: status ? logoObject.realImageUrl : '',
```

Comment:
What do you think if we don't send `realImageUrl` when `status` is false? Like, instead of sending an empty string, just don't send anything.

Something along the lines of:

```javascript
return {
  status,
  ...status && { realImageUrl: logoObject.realImageUrl }
}
```

Reply:
I tried this at first, but, don't you think having a consistent response is better?